rclpy/test/test_parameter.py

```diff
@@ -138,6 +139,16 @@ def test_integer_tuple_array(self):
             Parameter.Type.INTEGER_ARRAY, Parameter.Type.from_parameter_value(int_tuple))
         self.assertTrue(Parameter.Type.check(Parameter.Type.INTEGER_ARRAY, int_tuple))
 
+    def test_integer_array(self):
+        int_array = array('i', [1, 2, 3])
+        self.assertEqual(
+            Parameter.Type.INTEGER_ARRAY, Parameter.Type.from_parameter_value(int_array))
+
+    def test_double_array(self):
+        double_array = array('d', [1.0, 2.0, 3.0])
+        self.assertEqual(
+            Parameter.Type.DOUBLE_ARRAY, Parameter.Type.from_parameter_value(double_array))
+
```

Comment:
@BMarchi IIRC this feature request was the result of `Parameter.from_parameter_msg` failing. Is that right @gonzodepedro ? If so, a test for that would be nice :)